sys/usb/usbus/cdc/acm/cdc_acm.c

```diff
@@ -0,0 +1,304 @@
+    memset(&cdcacm->iface_ctrl, 0, sizeof(usbus_interface_t));
+    memset(&cdcacm->iface_data, 0, sizeof(usbus_interface_t));
+    /* Configure Interface 0 as control interface */
+    cdcacm->iface_ctrl.class = USB_CLASS_CDC_CONTROL ;
+    cdcacm->iface_ctrl.subclass = USB_CDC_SUBCLASS_ACM;
+    cdcacm->iface_ctrl.protocol = USB_CDC_PROTOCOL_NONE;
+    cdcacm->iface_ctrl.hdr_gen = &cdcacm->cdcacm_hdr;
+    cdcacm->iface_ctrl.handler = handler;
+    /* Configure second interface to handle data endpoint */
+    cdcacm->iface_data.class = USB_CLASS_CDC_DATA ;
+    cdcacm->iface_data.subclass = USB_CDC_SUBCLASS_NONE;
+    cdcacm->iface_data.protocol = USB_CDC_PROTOCOL_NONE;
+    cdcacm->iface_data.hdr_gen = NULL;
+    cdcacm->iface_data.handler = handler;
+
+    /* Create required endpoints */
+    cdcacm->ep_ctrl = usbus_add_endpoint(usbus, &cdcacm->iface_ctrl, USB_EP_TYPE_INTERRUPT, USB_EP_DIR_IN, 8);
+    cdcacm->ep_ctrl->interval = 255;
+    cdcacm->ep_data_out = usbus_add_endpoint(usbus, &cdcacm->iface_data, USB_EP_TYPE_BULK, USB_EP_DIR_OUT, 64);
+    cdcacm->ep_data_out->interval = 0;
+    cdcacm->ep_data_in = usbus_add_endpoint(usbus, &cdcacm->iface_data, USB_EP_TYPE_BULK, USB_EP_DIR_IN, 64);
+    cdcacm->ep_data_in->interval = 0;
+    /* Add interfaces to the stack */
+    usbus_add_interface(usbus, &cdcacm->iface_ctrl);
+    usbus_add_interface(usbus, &cdcacm->iface_data);
+
+    usbdev_ep_ready(cdcacm->ep_data_out->ep, 0);
+    usbus_enable_endpoint(cdcacm->ep_data_out);
+    usbus_enable_endpoint(cdcacm->ep_data_in);
+    usbus_enable_endpoint(cdcacm->ep_ctrl);
+    usbus_handler_set_flag(handler, USBUS_HANDLER_FLAG_RESET);
+}
+
+static int _setup_handler(usbus_t *usbus, usbus_handler_t *handler,
+                          usbus_setuprq_state_t state, usb_setup_t *setup)
+{
+    (void)state;
+    (void)usbus;
+    usbus_cdcacm_device_t *cdcacm = (usbus_cdcacm_device_t*)handler;
+    DEBUG("Request:0x%x\n", setup->request);
+    switch(setup->request) {
+        case USB_SETUP_REQ_SET_LINE_CODING:
+            DEBUG("Value:0x%x, interface:%d, len:%d\n", setup->value,
+                                                        setup->index,
+                                                        setup->length);
+            break;
+        case USB_SETUP_REQ_SET_CONTROL_LINE_STATE:
+            if (setup->value & USB_CDC_ACM_CONTROL_LINE_DTE) {
+                DEBUG("CDC ACM: DTE Enabled\n");
+                cdcacm->state = USBUS_CDCACM_LINE_STATE_DTE;
+                _cdc_acm_flush(cdcacm);
+            }
+            else {
+                cdcacm->state = USBUS_CDCACM_LINE_STATE_DISCONNECTED;
+                DEBUG("CDC ACM: DTE disabled\n");
+            }
+            DEBUG("Value:0x%x, interface:%d, nb:%d\n", setup->value,
+                                                       setup->index,
+                                                       setup->length);
+            break;
+        default:
```

Comment:
I don't recall if there is other setup request for ACM but isn't better to return -1 to send a stall ?

Reply:
You're right, added the `return -1`